notebook/nbextensions.py

```diff
@@ -199,60 +179,189 @@ def install_nbextension(path, overwrite=False, symlink=False, user=False, prefix
+                        logger=None):
+    """Install an nbextension bundled in a Python package.
+    
+    See install_nbextension for parameter information."""
+    m, nbexts = _get_nbextension_metadata(package)
+    base_path = os.path.split(m.__file__)[0]
+    for nbext in nbexts:
+        src = os.path.join(base_path, nbext['src'])
+        dest = nbext['dest']
+        require = nbext['require']
+        if logger:
+            logger.info(src, dest, require)
+        install_nbextension(src, overwrite=overwrite, symlink=symlink,
+            user=user, sys_prefix=sys_prefix, prefix=prefix, nbextensions_dir=nbextensions_dir,
+            destination=dest, logger=logger
+            )
+
+
+
+def uninstall_nbextension(dest, require, user=False, sys_prefix=False, prefix=None, 
+                          nbextensions_dir=None, logger=None):
+    """Uninstall a Javascript extension of the notebook
+    
+    Removes staged files and/or directories in the nbextensions directory and 
+    removes the extension from the frontend config.
+    
+    Parameters
+    ----------
+    
+    dest : str
+        path to file, directory, zip or tarball archive, or URL to install
+        name the nbextension is installed to.  For example, if destination is 'foo', then
+        the source file will be installed to 'nbextensions/foo', regardless of the source name.
+        This cannot be specified if an archive is given as the source.
+    require : str
+        require.js path used to load the extension
+    user : bool [default: False]
+        Whether to install to the user's nbextensions directory.
+        Otherwise do a system-wide install (e.g. /usr/local/share/jupyter/nbextensions).
+    prefix : str [optional]
+        Specify install prefix, if it should differ from default (e.g. /usr/local).
+        Will install to ``<prefix>/share/jupyter/nbextensions``
+    nbextensions_dir : str [optional]
+        Specify absolute path of nbextensions directory explicitly.
+    logger : Jupyter logger [optional]
+        Logger instance to use
+    """
+    nbext = _get_nbextension_dir(user=user, sys_prefix=sys_prefix, prefix=prefix, nbextensions_dir=nbextensions_dir)
+    dest = cast_unicode_py2(dest)
+    full_dest = pjoin(nbext, dest)
+    if os.path.lexists(full_dest):
+        if logger:
+            logger.info("Removing: %s" % full_dest)
+        if os.path.isdir(full_dest) and not os.path.islink(full_dest):
+            shutil.rmtree(full_dest)
+        else:
+            os.remove(full_dest)
+    
+    # Look through all of the config sections making sure that the nbextension
+    # doesn't exist.
+    config_dir = os.path.join(_get_config_dir(user=user, sys_prefix=sys_prefix), 'nbconfig')
+    cm = BaseJSONConfigManager(config_dir=config_dir)
+    for section in ['common', 'notebook', 'tree', 'edit', 'terminal']:
+        cm.update(section, {"load_extensions": {require: None}})
+
+def uninstall_nbextension_python(package,
```

Comment:
This function, and all others, should have complete docstrings explaining their parameters (like `uninstall_nbextension` above nicely does).

As a general rule, we shouldn't merge PRs whose docstrings don't document their parameters (I'm sure that's been violated in the past, but we should do better moving forward).